trusted/wallet.cpp

```diff
@@ -89,6 +94,46 @@ bool wallet::check_password(const password_t& mp_)
     return false;
 }
 
+bool wallet::check_password_policy(const password_t& mp_)
+{
+    if (mp_.size() < 8)
+    {
+        on_error("[create_wallet] Master password must be at least 8 characters long");
+        return false;
+    }
+    int num   = 0;
+    int punct = 0;
+    for (size_t i = 0; i < mp_.size(); i++)
+    {
+        if (std::isdigit(mp_[i]))
+        {
+            num++;
+        }
+
+        if (std::ispunct(mp_[i]))
+        {
+            punct++;
+        }
+
+        if (num > 0 && punct > 0)
```

Comment:
I think it would be better to let it go through the whole password.
Cutting it short might enable `timing_attack`.

Reply:
How do you do timing attack when creating the wallet